framework/wazuh/core/batcher/batcher.py

```diff
@@ -0,0 +1,196 @@
+
+        Raises
+        ------
+        Exception
+            If an error occurs during the sending of the buffer.
+        """
+        try:
+            async with get_indexer_client() as indexer_client:
+                list_of_uid: List[uuid.UUID] = [event.uid for event in events]
+                bulk_list: List[BulkDoc] = [
+                    BulkDoc.create(index=indexer_client.events.INDEX, doc_id=None, doc=event.msg) for event in events
+                ]
+
+                response = await indexer_client.events.bulk(data=bulk_list)
+
+                for response_item, uid in zip(response["items"], list_of_uid):
+                    response_item_msg = response_item["create"]
+                    response_msg = Message(uid=uid, msg=response_item_msg)
+                    self.q.send_to_demux(response_msg)
+        except Exception as e:
+            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+            logger.error(f"Error when sending buffer:\n{''.join(tb_str)}")
+
+    def create_flush_buffer_task(self):
+        """
+        Creates a task to flush the current buffer and reset it. This task sends the buffered messages to the indexer
+        and clears the buffer.
+        """
+        buffer_copy = self._buffer.copy()
+        asyncio.create_task(self._send_buffer(buffer_copy))
+        self._buffer.reset()
+
+    async def run(self):
+        """
+        Continuously retrieves messages from the queue and batches them based on the configuration. Handles signals
+        for shutdown and manages the batching logic.
+
+        Handles:
+        - Retrieving messages from the queue.
+        - Checking buffer limits.
+        - Creating and managing flush tasks.
+        - Handling shutdown signals.
+        """
+        # Register signal handlers
+        self._shutdown_event = asyncio.Event()
+        loop = asyncio.get_running_loop()
+        loop.add_signal_handler(signal.SIGINT, self._handle_signal, signal.SIGINT)
+        loop.add_signal_handler(signal.SIGTERM, self._handle_signal, signal.SIGTERM)
+
+        try:
+            while not self._shutdown_event.is_set():
+                done, pending = await asyncio.wait(
+                    [self._get_from_queue(), self._timer.wait_timeout_event()],
+                    return_when=asyncio.FIRST_COMPLETED
+                )
+
+                # Process completed tasks
+                for task in done:
+                    if isinstance(task.result(), Message):
+                        message = task.result()
+
+                        if self._buffer.get_length() == 0:
+                            self._timer.create_timer_task()
+
+                        self._buffer.add_message(message)
+
+                        if self._buffer.check_count_limit() or self._buffer.check_size_limit():
+                            self.create_flush_buffer_task()
+                            self._timer.reset_timer()
```

Comment:
Is the timer reset every time we create a flush buffer task? In that case, we could reset it inside the method `create_flush_buffer_task`

Reply:
Changed in 92c3106a91132f0a82fbc5b27062532886153f28.